app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -98,8 +119,32 @@ open class ProfilesViewModel(
           _profiles.value = profileList
           _filteredProfiles.value = filteredProfiles
           _loading.value = false
+          _isConnected.value = true
+          waitingDone.value = false
+          isWaiting.value = false
+          Log.e(
+              "ConnectionCheck",
+              "Reset flags: waiting=${isWaiting.value}, done=${waitingDone.value}")
         },
-        onFailure = { e -> handleError(e) })
+        onFailure = { e ->
+          Log.e("ConnectionCheck", "Request FAILED")
+          Log.e(
+              "ConnectionCheck",
+              "Current state: waiting=${isWaiting.value}, done=${waitingDone.value}")
+          handleError(e)
+          if (e is com.google.firebase.firestore.FirebaseFirestoreException) {
+            if (!waitingDone.value && !isWaiting.value) {
+              Log.e("ConnectionCheck", "First failure, starting timer")
+              checkConnectionWithDelay()
+              isWaiting.value = true
+            }
+
+            if (waitingDone.value && isWaiting.value) {
+              Log.e("ConnectionCheck", "Timer completed and still failing, showing offline")
```

Comment:
Recommendation: Consider changing this to Log.i since it’s an informational log about the state, but it's noteworthy enough to merit visibility.

Reply:
same here